pkg/storage/types/pvc.go

```diff
@@ -256,3 +256,37 @@ func RenderPVC(size *resource.Quantity, claimName, namespace, storageClass, acce
 
 	return pvc
 }
+
+func IsHotplugVolume(vol *virtv1.Volume) bool {
+	if vol == nil {
+		return false
+	}
+	volSrc := vol.VolumeSource
+	if volSrc.PersistentVolumeClaim != nil && volSrc.PersistentVolumeClaim.Hotpluggable {
+		return true
+	}
+	if volSrc.DataVolume != nil && volSrc.DataVolume.Hotpluggable {
+		return true
+	}
+	if volSrc.MemoryDump != nil && volSrc.MemoryDump.PersistentVolumeClaimVolumeSource.Hotpluggable {
+		return true
+	}
+
+	return false
+}
+
+func GetVolumesByName(vmiSpec *virtv1.VirtualMachineInstanceSpec) map[string]*virtv1.Volume {
```

Comment:
This can be used in some places in the code base if you change the signature to array of volumes.

Reply:
@xpivarc I need a map because I don't want to iterate multiple times over the volumes again and again. In this case, this is used for getting the old volumes that might be in a different order then the new ones.
But I'm sure we can refactor other part of the codes to use this function

Reply:
Let's follow up